ophyd/v2/_channelsim.py

```diff
@@ -23,9 +24,16 @@ def make_sim_descriptor(source: str, value) -> Descriptor:
         dtype = primitive_dtypes[type(value)]
         shape = []
     except KeyError:
-        assert isinstance(value, Sequence), f"Can't get dtype for {type(value)}"
-        dtype = "array"
-        shape = [len(value)]
+        if isinstance(value, Sequence):
+            dtype = "array"
+            shape = [len(value)]
+        elif isinstance(value, Enum):
+            dtype = "string"
+            shape = []
+            choices = [e.value for e in type(value).__members__.values()]
```

Comment:
As we can list enums this can be simplified then
```suggestion
            choices = [e.value for e in type(value)]
```